pretyping/pretyping.ml

```diff
@@ -186,9 +186,23 @@ type inference_flags = {
 let frozen_holes (sigma, sigma') =
   (); fun ev -> Evar.Map.mem ev (Evd.undefined_map sigma)
 
+let rec advance sigma evk =
```

Comment:
I think `advance` should go to `Evarutils` rather than here. But as you (@herbelin) mention above, this should replace, not duplicate, the code in `Proofview`.